tests/functional/pv/pv_services/test_selinux_with_rwop.py

```diff
@@ -0,0 +1,362 @@
+
+    @pytest.mark.parametrize("copies", [5])
+    def test_selinux_relabel_on_rwop(
+        self, pvc_factory, service_account_factory, copies
+    ):
+        """
+        Steps:
+            1. Create cephfs RWOP and RWO pvc and pod with 100K files across multiple nested directories
+            2. Observe the mountflags in csi-cephfsplugin-xx pod csi-cephfsplugin container logs
+            3. Take md5sum for random files in pod
+            4. Take backup of each pod's yaml.
+            5. Change `securityContext.SeLinuxOptions.Level: s0:c26,c20` value to something different on rwop pod
+            and apply the new pod yaml
+            6. Observe the new mountflags
+
+        Args:
+            pvc_factory (function): A call to pvc_factory function
+            service_account_factory (function): A call to service_account_factory function
+            copies (int): number of copies to write kernel files in pod
+
+        """
+
+        # Create cephfs rwop pvc
+        self.pvc_obj = pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            access_mode=constants.ACCESS_MODE_RWOP,
+            size="20",
+        )
+        self.ocp_project = self.pvc_obj.project
+
+        # Create service_account to get privilege for deployment pods
+        self.service_account_obj = service_account_factory(
+            project=self.pvc_obj.project,
+        )
+
+        # Create deployment pod
+        self.pod_obj = self.create_deploymentconfig_pod(
+            command=["/opt/multiple_files.sh"],
+            command_args=[f"{copies}", "/mnt"],
+        )
+        log.info(f"files copied to pod {self.pod_obj.name}")
+
+        expected_log = "system_u:object_r:container_file_t"
+        pod_name = self.get_csi_cephfs_pod_on_same_node(self.pod_obj)
+
+        log.info(f"Check logs of csi-cephfsplugin-xxx/csi-cephplugin pods {pod_name}")
+        sample = TimeoutSampler(
+            timeout=100,
+            sleep=5,
+            func=verify_log_exist_in_pods_logs,
+            pod_names=[pod_name],
+            container="csi-cephfsplugin",
+            expected_log=expected_log,
+            all_containers_flag=False,
+            since="120s",
+        )
+        if not sample.wait_for_func_status(result=True):
+            raise ValueError(
+                f"The expected log '{expected_log}' does not exist in {pod_name}"
+            )
+
+        # Leave pod for some time to run since file creation time is longer
+        waiting_time = 10
+        log.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
+
+        # Backup file
+        self.backup_pod(pod_obj=self.pod_obj)
+        pod_attach_time_1 = self.get_pod_start_time(self.pod_obj.name)
+
+        log.info(f"Time taken by pod to restart is {pod_attach_time_1}")
+        log.info(
+            f"Sleep 120 seconds so the logs in {self.pod_obj.name} will be updated"
+        )
+        time.sleep(120)
```

Comment:
Do you wait for a specific log? if yes, you can use the generator `TimeoutSampler`